lib/phase/functional.js

```diff
@@ -254,12 +254,29 @@ module.exports = async ({
     const doc = flattenedDoc;
     let errors = [];
     let warnings = [];
+    let pipelineOnly = true;
 
     // Jobs
     errors = errors.concat(validateJobSchema(doc));
 
+    if (doc.stages) {
+        pipelineOnly = false;
+    }
```

Comment:
Can we abstract this check into workflow parser (i.e. `generateWorkflowGraph`) as the entire doc is being passed into to it?